python/hail/expr/functions.py

```diff
@@ -184,6 +184,66 @@ def cond(predicate, then_case, else_case):
     return construct_expr(Condition(predicate._ast, then_case._ast, else_case._ast),
                           then_case._type, indices, aggregations, joins)
 
+def bind(expr, f):
+    """Bind a temporary variable and use it in a function.
+
+    Examples
+    --------
+    Expressions are "inlined", leading to perhaps unexpected behavior
+    when randomness is involved. For example, let us define a variable
+    `x` from the :meth:`rand_unif` method:
+
+    >>> x = functions.rand_unif(0, 1)
+
+    Note that evaluating `x` multiple times returns different results.
+    The value of evaluating `x` is unknown when the expression is defined.
+
+    .. doctest::
+
+        >>> eval_expr(x)
+        0.3189309481038456
```

Comment:
I don't know how doctest works; is it going to fail the test if `eval_expr(x)` doesn't equal `0.3189309481038456`?